[junyi73] iP

junyi73 opened the pull request:
## DukePro
> “Your mind is for having ideas, not holding them.” – David Allen([source](https://dansilvestre.com/productivity-quotes))

DukePro frees your mind of having to remember things you need to do. It's,
- text-based
- easy to learn 
- ~~FAST~~ _ SUPER_ FAST to use

All you need to do is,

1. Download it from [here](google.com).
2. double click it.
3. add your tasks.
4. let it manage your tasks for you :wink:

And it is **FREE**!

Features:

- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:

```
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

junyi73 committed:
feat: update chatbot name and welcome message

junyi73 committed:
feat: added echo feature and leave feature

junyi73 committed:
feat: added list feature

junyi73 committed:
feat: added markDone feature

junyi73 committed:
feat: added support for tracking of three types of tasks

junyi73 committed:
feat: added automated testing

junyi73 committed:
feat: added ability to handle errors

junyi73 committed:
fix: fixed bug

junyi73 committed:
feat: added ability to delete tasks

junyi73 committed:
Add save feature

junyi73 committed:
Merge branch 'branch-Level-7'

junyi73 committed:
add TaskList class

junyi73 committed:
Add Dates and Times

junyi73 committed:
add additional classes for encapsulation

junyi73 committed:
move classes under noisy package

junyi73 committed:
Merge branch 'add-gradle-support'

junyi73 committed:
create a JAR file

reidenong commented on src/main/java/Noisy/Parser.java:
```diff
@@ -0,0 +1,47 @@
+        return LocalDate.parse(date);
+    }
+
+    public int parseInput(String input, TaskList taskList) {
+        if (input.equals("bye")) {
+            return 0;
+        }
+
+        if (input.equals("list")) {
+            return 1;
+        }
+
+        if (input.startsWith("mark")) {
+            return 2;
+        }
+
+
+        Task task = null;
+        try {
+            switch (input.split(" ")[0]) {
+                case "todo":
+                    if (input.split(" ", 2).length < 2) {
+                        throw new NoisyException("OOPS!!! The description of a todo cannot be empty.");
+                    }
+                    return 3;
+                case "Noisy.Deadline":
+                    return 4;
+                case "Noisy.Event":
+                    return 5;
+                case "delete":
+                    return 6;
+                default:
+                    throw new NoisyException("OOPS!!! I'm sorry, but I don't know what that means :-(");
+            }
+        } catch (NoisyException e) {
+            System.out.println(e);
+        }
+        return 7;
+    }
```
Is this parseInput() function used by the main class to parse the user input?

reidenong commented on src/main/java/Noisy/Noisy.java:
```diff
@@ -0,0 +1,75 @@
+            if (input.startsWith("mark ")) {
+                String[] string = input.split(" ");
+                Integer index = Integer.parseInt(string[1]);
+                taskList.markDoneFromList(index - 1);
+                ui.printMark(index, taskList);
+                continue;
+            }
+
+            task = null;
+            try {
+                switch (input.split(" ")[0]) {
+                    case "todo":
+                        if (input.split(" ", 2).length < 2) {
+                            throw new NoisyException("OOPS!!! The description of a todo cannot be empty.");
+                        }
+                        task = new Todo(input.split(" ", 2)[1], false);
+                        break;
+                    case "Deadline":
+                        input = input.trim();
+                        String[] string = input.split(" ", 4);
+                        task = new Deadline(string[1], Boolean.parseBoolean(string[2]), parser.parseDate(string[3]));
+                        break;
+                    case "Event":
+                        String[] eventString = input.split(" ", 5);
+                        task = new Event(eventString[1], false, parser.parseDate(eventString[3]), parser.parseDate(eventString[4]));
+                        break;
+                    case "delete":
+                        String[] deleteString = input.split(" ");
+                        Integer index = Integer.parseInt(deleteString[1]);
+                        Task deletedTask = taskList.getTask(index - 1);
+                        taskList.deleteFromList(index - 1);
+                        ui.printDelete(index - 1, taskList);
+                        continue;
+                    default:
+                        throw new NoisyException("OOPS!!! I'm sorry, but I don't know what that means :-(");
+                }
+            } catch (NoisyException e) {
+                System.out.println(e);
+                continue;
+            }
```
Maybe we could use the parser class here to improve the flow of input?

reidenong commented on src/test/java/Noisy/TodoTest.java:
```diff
@@ -0,0 +1,12 @@
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class TodoTest {
+    @Test
+    public void dummyTest() {
+        assertEquals(2, 2);
+    }
```
Maybe we should include more test cases?

reidenong reviewed (commented):
Code seems functional.

Might be better if we had more comprehensive unit tests.

MingYiAw commented on src/main/java/Noisy/Noisy.java:
```diff
@@ -0,0 +1,75 @@
+package Noisy;
```
Should the package name be in lowercase?

MingYiAw commented on src/main/java/Noisy/Deadline.java:
```diff
@@ -0,0 +1,24 @@
+
+    @Override
+    public String formatTask() {
+        String status = isDone? "1" : "0";
```
Perhaps adding a space before the question mark here? Noticed this in several other places too.

MingYiAw commented on src/main/java/Noisy/Noisy.java:
```diff
@@ -0,0 +1,75 @@
+                        ui.printDelete(index - 1, taskList);
+                        continue;
+                    default:
+                        throw new NoisyException("OOPS!!! I'm sorry, but I don't know what that means :-(");
```
Should the `case` and `default` have indentation?

MingYiAw commented on src/main/java/Noisy/Storage.java:
```diff
@@ -0,0 +1,68 @@
+import java.util.ArrayList;
+import java.util.Scanner;
+import java.io.FileWriter;
+import java.io.BufferedWriter;
```
Perhaps ordering the imports in alphabetical order?

MingYiAw commented on src/main/java/Noisy/Storage.java:
```diff
@@ -0,0 +1,68 @@
+                file.createNewFile();
+                return tasks;
+
+            }Scanner scanner = new Scanner(file);
```
Should the `Scanner scanner = new Scanner(file);` be moved to next line?

MingYiAw commented on src/main/java/Noisy/Task.java:
```diff
@@ -0,0 +1,29 @@
+
+public abstract class Task {
+    protected String description;
+    protected boolean isDone;
```
I like how you name booleans following the coding standard. i.e starts with "is".

MingYiAw commented on src/main/java/Noisy/Todo.java:
```diff
@@ -0,0 +1,19 @@
+
+
+    @Override
+    public String formatTask() {
```
Perhaps there is an extra empty line here? I noticed the same issue in several other places too.

MingYiAw commented on src/main/java/Noisy/Event.java:
```diff
@@ -0,0 +1,27 @@
+    }
+
+    @Override
+    public String formatTask() {
```
I like how you name methods as a verb that reflect what the method does, which comply with the coding standard.

MingYiAw commented on src/main/java/Noisy/Ui.java:
```diff
@@ -0,0 +1,59 @@
+public class Ui {
+
+    public void printWelcome() {
+        String welcomeMessage = "____________________________________________________________\n"
+                + " Hello! I'm Noisy.Noisy\n"
+                + " What can I do for you?\n"
+                + "____________________________________________________________\n"
+                + " Bye. Hope to see you again soon!\n"
+                + "____________________________________________________________";
```
I like how you break the string into a few lines, which increase the readability.

MingYiAw reviewed (commented):
LGTM. Might require some minor changes in a few parts.

junyi73 committed:
Add JUnit tests

junyi73 committed:
Add JavaDoc headers

junyi73 committed:
Update code style to comply with coding standards

junyi73 committed:
add find feature

junyi73 committed:
Merge branch 'branch-Level-9'

junyi73 committed:
Merge branch 'branch-A-JavaDoc'

# Conflicts:
#	src/main/java/noisy/Task.java

junyi73 committed:
Revert "Merge branch 'branch-A-JavaDoc'"

This reverts commit b35f6107e1f998c9cc6e6e0ffead2fdab383138c, reversing
changes made to 5d1ecc34f33505541695986a05a9c9f34ec04c07.

junyi73 committed:
Merge branch 'branch-A-JavaDoc' of https://github.com/junyi73/ip into branch-A-JavaDoc

# Conflicts:
#	src/main/java/noisy/Task.java

junyi73 committed:
Merge branch 'branch-A-JavaDoc'

# Conflicts:
#	src/main/java/noisy/Task.java

junyi73 committed:
resolve merge conflict

junyi73 committed:
Merge branch 'branch-A-JavaDoc'

# Conflicts:
#	src/main/java/Noisy/Deadline.java
#	src/main/java/Noisy/Event.java
#	src/main/java/Noisy/TaskList.java
#	src/main/java/Noisy/Todo.java
#	src/main/java/noisy/Noisy.java
#	src/main/java/noisy/Ui.java

junyi73 committed:
add find feature

junyi73 committed:
set up javafx

junyi73 committed:
add GUI

junyi73 committed:
Add assertions to validate key assumptions in Parser class

The `Parser` class is responsible for processing user input and performing actions such as adding tasks, marking tasks as done, and deleting tasks from the task list. However, several important assumptions in the code are not explicitly documented, which could lead to runtime errors or incorrect behavior if the input or task operations are invalid.

This commit introduces the use of Java's `assert` feature to document and enforce key assumptions at various points in the code, such as:
- Ensuring that method parameters (input, taskList, storage, and ui) are not null.
- Verifying that string inputs like date and command formats are correctly structured.
- Asserting that task indices (for marking or deleting tasks) are within valid bounds of the task list.
- Checking that a task is not null before adding it to the task list.

junyi73 committed:
Merge branch 'branch-Level-10'

junyi73 committed:
Improve Code Quality of Parser and Storage Class

- Improved overall code organization and maintainability.
- Refactored variable names to be more accurate.
- Added egyptian braces to some methods

junyi73 committed:
Remove unused import statements, variables. Remove redundant strings.

junyi73 committed:
Merge pull request #2 from junyi73/branch-A-Assertions

Add assertions to validate key assumptions in Parser class

junyi73 committed:
Merge branch 'master' of https://github.com/junyi73/ip

junyi73 committed:
Merge branch 'master' into branch-A-CodeQuality

junyi73 committed:
Merge pull request #3 from junyi73/branch-A-CodeQuality

Branch a code quality

junyi73 committed:
Merge branch 'master' of https://github.com/junyi73/ip

junyi73 committed:
add snooze feature

junyi73 committed:
Merge branch 'branch-B-Snooze'

junyi73 committed:
update README.md

junyi73 committed:
rename Ui.png

junyi73 committed:
Fix null task saving to storage bug

junyi73 committed:
Fix syntax for Deadline, Event

junyi73 committed:
Fix delete indexing bug

junyi73 committed:
Fix loading event tasks from storage bug

junyi73 committed:
Fix delete command bugs

junyi73 committed:
Fix snooze command bugs

junyi73 committed:
Update user guide

junyi73 committed:
Update user guide

junyi73 committed:
Merge branch 'branch-BugFixes'
